### ISSUE TITLE
Moving items in multiple frames removes them from their frames

### ISSUE THREAD

**Issue by ThomWright** [excalidraw/excalidraw#6847]
## Steps

1. Make two frames with shapes inside
2. Select a shape in each frame
3. Move the selected shapes within their frames

## Expected behaviour

The shapes stay included in their respective frames.

## Actual behaviour

Only one shape stays included.

In the video below, the circle in the stop frame stops being included in the frame.

https://github.com/excalidraw/excalidraw/assets/1884486/6f8ac79b-1b0f-44a2-90a7-59475716b172

I believe I have seen other behaviour which ends up with similar results: shapes being excluded from frames they were in.

**Comment by dwelle** [comment]
This is actually by design. When dragging elements, they're added to the frame under cursor. If the select element is outside the frame, we don't add it because it would become invisible.

Dragging elements that are in different frames seems like too much an edge case to write specific rules for this.

**Comment by ryan-di** [comment]
When dragging elements to the frame, the logic is this:
- for the dragged elements, when the cursor is in the frame (which would lead to the highlighting of the target frame), the ones that at least intersect with the frame are added to the target frame
- the ones that are not in the target frame are removed from it (by setting `element.frameId` to `null`, which is why element from "frame 34" was removed from it in your clip)

I suppose, for the elements that were in some other frames, if they are not in the target frame on pointer up, they could just be kept in their original frame if they at least intersect with original frame. But then again, the logic gets a bit more complicated if they are grouped elements, which we deem as "atomic" elements. Luckily, we have all the relevant helpers already.

And I guess when trying to align items from different frames by moving them together, the current behavior can be annoying. I'm happy to take a look at it when I'm a bit more available :) cheers!

**Comment by ThomWright** [comment]
That's good to hear, thanks 🙂

My use case right now is preparing slides for a presentation in Google Slides. I have a diagram, which I build up a bit at a time.

I'm using a frame per slide, and sometimes I want to rearrange something. It's important to keep the relative positions consistent within each frame, otherwise parts of the diagram would jump around between slides.

As an example, these three frames would be three slides.

<img width="313" alt="image" src="https://github.com/excalidraw/excalidraw/assets/1884486/59187c54-cf10-433a-ba06-5f8c55910650">

Using frames make it easier to organise within Excalidraw, keeping a consistent bounding box around the diagrams as elements appear/disappear, and being able to move whole frames together.

Loving the frames!

**Comment by ryan-di** [comment]
@ThomWright that's a very valid use case, thanks! moving people together away from the bouncer, gotta support that :p

**Comment by ThomWright** [comment]
Some other things which seem to remove (or not add) elements from frames:

- Pasting in an element into a frame – not added as I would expect
- Using the alignment tool within a frame – elements can be removed from the frame, despite staying within the bounds

**Comment by ryan-di** [comment]
Yeah the first point is by design and the second point has been tracked here https://github.com/excalidraw/excalidraw/issues/6851. Will have another review regarding the first point 👍

**Comment by taotie111** [comment]
is there anyone support for that? if there is no one, I will do it.

**Comment by DidasMba** [comment]
Hi @ThomWright  

I would like to pick up this issue (#6847) and work on keeping elements in their original frames when moving multiple items. 

Thank you!

**Comment by jwgarcia** [comment]
Hi @ThomWright Is @DidasMba working on this issue or could I pick it up? 

Thanks!!

**Comment by Rafael-Duarte-Silva** [comment]
Hi, I hope this helps.
